oligo_designer_toolsuite/oligo_specificity_filter/_filter_base.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import os
+import re
+import sys
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+import joblib
+import pandas as pd
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+
+from ..IO._data_parser import write_oligos_DB_tsv
+
+
+class ProbeFilterBase(ABC):
+    def __init__(
+        self,
+        n_jobs,
+        dir_output,
+        dir_annotations=None,
+        number_subbatches=None,
+        max_genes_in_batch=300,
+    ):
+        """This is the base class for all probe filter classes
+
+        :param n_jobs: numbers of jobs to run in parallel when filtering the probes
+        :type n_jobs: int
+        :param dir_output: output directory to write file containing filtered probes to
+        :type dir_output: str
+        :param dir_annotations: directory storing file annotations of probes, defaults to None
+        :type dir_annotations: str, optional
+        :param number_subbatches: number of subbatches to split batched data into to control how many genes are in one batch, defaults to None
+        :type number_subbatches: int, optional
+        :param max_genes_in_batch: max number of genes allowed in a batch, defaults to 300
+        :type max_genes_in_batch: int, optional
+        """
+        self.dir_output = dir_output
+        self.max_genes_in_batch = max_genes_in_batch  # if more than 300 genes in one batch split into subbatches to reduce required memory for loading blast results
+        self.duplicated_sequences = None
+        self.number_subbatches = number_subbatches
+
+        if n_jobs == None:
+            self.n_jobs = joblib.cpu_count()
+        else:
+            self.n_jobs = n_jobs
+
+        # set logger
+        log_file = os.path.join(self.dir_output, "logs/filter_probes.log")
```

Comment:
we should think about this logging situation. do we need multiple loggers? If yes we should write a seperate function for logger setup